### PR TITLE
Max pool2d ceil mode to tosa

### PR DESCRIPTION
**Pull request by lisaliu1**
This PR implements the support for lowering torch.aten.max_pool2d to tosa with ceil_mode==true.

### CODE REVIEW

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 1]
```diff
@@ -4030,23 +4047,22 @@ static LogicalResult getOutputTypeAndPoolingParameters(
...
+  
   bool ceilMode;
   if (!matchPattern(op.getCeilMode(), m_TorchConstantBool(&ceilMode)))
     return rewriter.notifyMatchFailure(
         op, "only support constant bool ceil_mode for pooling op");
-  if (ceilMode)
-    return rewriter.notifyMatchFailure(
-        op, "only support ceil_mode equals to False for pooling op");
 
+  // add ceil_mode support.
```
Comment (eric-k256) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:4059]: nit: this comment only makes sense for this change, I would delete it.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 2]
```diff
@@ -3977,17 +3987,24 @@ template <typename AtenOpT, typename tosaOp>
...
 
   int64_t outputHDim = ConvertAtenPoolingBaseOp<AtenOpT, tosaOp>::getOutputDim(
       inputShape[inputRank - 2], kernelSize[0], strideArray[0], padArray[0],
-      padArray[0], dilationArray[0]);
+      padArray[0], dilationArray[0], ceilMode);
   int64_t outputWDim = ConvertAtenPoolingBaseOp<AtenOpT, tosaOp>::getOutputDim(
       inputShape[inputRank - 1], kernelSize[1], strideArray[1], padArray[1],
-      padArray[1], dilationArray[1]);
+      padArray[1], dilationArray[1], ceilMode);
+  padArray[0] = (outputHDim - 1) * strideArray[0] +
```
Comment (eric-k256) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:4002]: Code formatting looks a bit inconsistent here (sometimes a*b and others a * b). Can you verify you've run clang-format? `git clang-format HEAD~1` should only modify lines you have already changed.
https://github.com/llvm/torch-mlir/wiki/Coding-Style

**Review by eric-k256 — COMMENTED** [review]
Your design looks right, just two minor points to check.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp (hunk 2 above)
Comment (lisaliu1) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp]: Thanks Eric for your reviews. I've addressed your comments.

**Review by eric-k256 — APPROVED** [review]
Thanks Lisa! Looks good to me.

**Pull request merged**